### PR TITLE
HUE-9382 Possible Script Execution on Add/Sync LDAP users (asnaik)

### PR DESCRIPTION
**Pull request by Akhilsnaik**
HUE-9382 Possible Script Execution on Add/Sync LDAP users (asnaik) Fix

Tested in : Safari , Chrome & Firefox 

image

### CODE REVIEW

**Comment by Akhilsnaik** [comment]
@JohanAhlen  @romainr  , 
Please help me review this.

**Comment by sreenaths** [comment]
Instead of an entityMap, a better implementation to sanitise might be to use innerText and innerHTML.
https://remarkablemark.org/blog/2019/11/29/javascript-sanitize-html/

File: apps/useradmin/src/useradmin/templates/layout.mako [hunk 1]
```diff
@@ -113,7 +129,7 @@ def is_selected(section, matcher):
           $el.closest('.control-group').addClass('error');
           var html = '<span class="help-inline"><ul class="errorlist">';
           e.message.forEach(function (message) {
-            html += '<li>' + message + '</li>';
+            html += '<li>' + escapeHtml(message) + '</li>';
```
Comment (sreenaths) [inline apps/useradmin/src/useradmin/templates/layout.mako:132]: What about sanitising before sending it into the server?

File: apps/useradmin/src/useradmin/templates/layout.mako [hunk 2]
```diff
@@ -104,6 +104,22 @@ def is_selected(section, matcher):
...
+     '/': '&#x2F;',
+     '`': '&#x60;',
+     '=': '&#x3D;'
+    };
+
+    function escapeHtml(string) {
+       return String(string).replace(/[&<"'`=\/]/g, function(s) {
+        return entityMap[s];
+       });
+    }
```
Comment (JohanAhlen) [inline apps/useradmin/src/useradmin/templates/layout.mako:122]: Use hueUtils.deXSS(str) instead

Comment (Akhilsnaik) [reply apps/useradmin/src/useradmin/templates/layout.mako]: Code changed as per review comment

Comment (Akhilsnaik) [reply apps/useradmin/src/useradmin/templates/layout.mako]: i have changed it to : hueUtils.escapeOutput

**Comment by romainr** [comment]
Nice!

**Pull request merged**